src/VisualStudio/Core/Def/Implementation/LanguageClient/RazorInProcLanguageClient.cs

```diff
@@ -25,8 +25,16 @@ namespace Microsoft.CodeAnalysis.ExternalAccess.Razor.Lsp
     /// TODO - This can be removed once C# is using LSP for diagnostics.
     /// https://github.com/dotnet/roslyn/issues/42630
     /// </summary>
+    /// <remarks>
+    /// This specifies RunOnHost because in LiveShare we don't want this to activate on the guest instance
+    /// because LiveShare drops the ClientName when it mirrors guest clients, so this client ends up being
+    /// activated solely by its content type, which means it receives requests for normal .cs and .vb files
+    /// even for non-razor projects, which then of course fails because it gets text sync info for documents
```

Comment:
That's a long sentence!

Might be good to add a bit saying that even without text sync (and even if client name was mirrored) we don't really want to mirror any requests from the guest to this LSP server - the generated razor text buffers only exist on the host so only the host LSP client should be making requests to it (via the razor LSP server).